packages/next/server/app-render.tsx

```diff
@@ -1002,18 +1051,34 @@ export async function renderToHTMLOrFlight(
         serverCSSForEntries,
         layoutOrPagePath
       )
-      const Template = template
-        ? await interopDefault(template())
-        : React.Fragment
+
+      const Template =
+        (await createFragmentTypeWithStyles({
+          filePath,
+          getComponent: template,
+          shouldPreload: true,
+        })) || React.Fragment
       const ErrorComponent = error ? await interopDefault(error()) : undefined
-      const Loading = loading ? await interopDefault(loading()) : undefined
+      const errorStyles =
+        filePath && error
+          ? await createFragmentTypeWithStyles({
+              filePath,
+              styleOnly: true,
+            })
+          : undefined
```

Comment:
Here `ErrorComponent` is a module reference, and required to be serializable. So we can't make a new FC (`() => <><Comp/>{styles}</>`) here.